pkg/services/accesscontrol/scope.go

```diff
@@ -142,19 +142,26 @@ func (s scopeProviderImpl) GetResourceAllIDScope() string {
 	return GetResourceAllIDScope(s.root)
 }
 
-// WildcardsFromPrefix generates valid wildcards from prefix
-// datasource:uid: => "*", "datasource:*", "datasource:uid:*"
 func WildcardsFromPrefix(prefix string) Wildcards {
+	return WildcardsFromPrefixes([]string{prefix})
+}
+
+// WildcardsFromPrefixes generates valid wildcards from prefixes
+// datasource:uid: => "*", "datasource:*", "datasource:uid:*"
+func WildcardsFromPrefixes(prefixes []string) Wildcards {
 	var b strings.Builder
 	wildcards := Wildcards{"*"}
-	parts := strings.Split(prefix, ":")
-	for _, p := range parts {
-		if p == "" {
-			continue
+	for _, prefix := range prefixes {
+		parts := strings.Split(prefix, ":")
+		for _, p := range parts {
+			if p == "" {
+				continue
+			}
+			b.WriteString(p)
+			b.WriteRune(':')
+			wildcards = append(wildcards, b.String()+"*")
```

Comment:
```suggestion
			wildcards = append(wildcards, fmt.Sprintf("%s:*", p))
```

Wouldn't this be equivalent if the builder is being reset every pass?

Reply:
It gets reset after we iterate over the parts. For instance, if you pass in the prefix `datasource:uid`, in the first iteration you'll produce `datasource:*`, and in the second `datasource:uid:*`. Using `fmt.Sprintf("%s:*", p)` would produce `datasource:*` and `uid:*`.

Reply:
just saw the reset is in the outer `for`, sorry :smile: